tools/bazelify_tests/grpc_repo_archive.sh

```diff
@@ -0,0 +1,118 @@
+#!/bin/bash
+# Copyright 2023 The gRPC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+ARCHIVE_WO_SUBMODULES="$1"
+ARCHIVE_WITH_SUBMODULES="$2"
+export ARCHIVES_DIR="$(pwd)/archives"
+
+export ARCHIVE_FORMAT=tar
+
+mkdir -p "${ARCHIVES_DIR}"
+rm -rf "${ARCHIVES_DIR}"/*
+
+# HACK: To be able to collect all grpc source files as an archive
+# we need to break from bazel's "sandbox" to be able to read files
+# from the original bazel workspace (which in our case is the grpc repository root)
+# This action runs with the "standalone" (a.k.a) local strategy,
+# so path to the original bazel workspace from where this was invoked
+# can be obtained by resolving the link that points to one of the
+# source files.
+# 1. find first component of the relative path to this script
+# 2. resolve the symlink (it will point to same dir in the workspace)
+# 3. one level up is the root of the original bazel workspace
+FIRST_PATH_COMPONENT="$(echo $0 | sed 's|/.*||')"
+ORIGINAL_BAZEL_WORKSPACE_ROOT="$(dirname $(readlink ${FIRST_PATH_COMPONENT}))"
+
+# extract STABLE_GIT_COMMIT from stable-status.txt
+GRPC_GIT_COMMIT_FROM_STABLE_STATUS=$(grep ^STABLE_GRPC_GIT_COMMIT bazel-out/stable-status.txt | cut -d' ' -f2)
+
+if [ "${GRPC_GIT_COMMIT_FROM_STABLE_STATUS}" == "" ]
+then
+  echo "Failed to obtain info about gRPC git commit from the bazel workspace. Make sure you invoke bazel with --workspace_status_command=tools/bazelify_tests/workspace_status_cmd.sh" >&2
+  exit 1
+fi
+
+GRPC_UNCOMMITED_PATCH_CHECKSUM_FROM_STABLE_STATUS=$(grep ^STABLE_GRPC_UNCOMMITED_PATCH_CHECKSUM bazel-out/stable-status.txt | cut -d' ' -f2)
+GRPC_GIT_WORKSPACE_DIRTY_FROM_STABLE_STATUS=$(grep ^STABLE_GRPC_GIT_WORKSPACE_DIRTY bazel-out/stable-status.txt | cut -d' ' -f2)
+
+pushd ${ORIGINAL_BAZEL_WORKSPACE_ROOT} >/dev/null
+
+if [ "${GRPC_GIT_COMMIT_FROM_STABLE_STATUS}" != "$(git rev-parse HEAD)" ]
+then
+  echo "Info about gRPC commit from stable-status.txt doesn't match the commit at which the workspace root is." >&2
```

Comment:
No particular situation in mind, but I thought there could be situations in which stable-status.txt could get out of date and if the "STABLE_*" values in stable-status.txt don't match the actual status of the git workspace, we could fail to recreate the archives when needed and that's a bad situation to be in since it will essentially test wrong revision of grpc and it can be hard to detect that. So better safe than sorry and I added an extra check for this.